src/lib/validators.ts

```diff
@@ -13,3 +13,13 @@ export const unixTimeFromDateString = new t.Type<number, string, unknown>(
     }),
   a => new Date(a).toISOString(),
 );
+
+export const validateEmail = (value: string) => {
+  if (value.length > 320) {
+    return false;
+  }
+
+  return /^[a-zA-Z0-9!#$%&'*+\-/=?^_`.{|}~]{1,64}@[a-z0-9.-]+\.[a-z]{2,64}$/.test(
+    value,
+  );
+};
```

Comment:
Email could be a branded string : P. This is quite cryptic but what I would personally like to have https://github.com/gcanti/io-ts/blob/master/index.md#branded-types--refinements